lvm: reuse existing volume group

brauner opened the pull request:
Closes #2914.

```
Reuse an already existing LVM volume group:
- lxc storage create <pool_name> lvm lvm.vg_name=LXDvg lvm.thinpool_name=LXDthin

Create a new LVM volume group or reuse an existing one:
- lxc storage create <pool_name> lvm source=/dev/sdb lvm.vg_name=LXDvg lvm.thinpool_name=LXDthin
- lxc storage create <pool_name> lvm source=/dev/sdb
```



@stgraber, do the semantics make sense to you?

brauner commented:
@stgraber, once you're fine with the semantics we'll likely need some upgrade code because I'm deprecating:

- `volume.lvm.thinpool_name` as a pool property propagating to a corresponding storage volume property and replace it by `lvm.thinpool_name` as a pool property that does not propagate.
- `lvm.thinpool_name` as a storage volume property inheriting from the aforementioned pool property.

Also, I am introducing the storage pool property:
- `lvm.vg_name` which corresponds to the already deprecated daemon configuration key `storage.lvm_vg_name`.

brauner commented:
Also, jenkins refuses to run on the branch. :)

brauner commented:
jenkins: test this please.

stgraber commented:
So with this, can I do:

    lxc storage create blah lvm source=my-vg

And end up with something that works? I'm hoping we can keep things somewhat aligned with what we've already done for ZFS.

Recording lvm.vg_name sounds fine for the same reason that zfs.pool_name is fine.

stgraber commented:
jenkins: test this please

stgraber commented:
@brauner jenkins looks unhappy with your change

brauner commented:
> So with this, can I do:

    lxc storage create blah lvm source=my-vg

> And end up with something that works? I'm hoping we can keep things somewhat aligned with what we've already done for ZFS.

@stgraber , yes, in case `my-vg` exists, `LXD` will set `source=my-vg` and `lvm.vg_name=my-vg`. In fact, `source` and `lvm.vg_name` will always e aligned once the volume group is either created or already exists. For example, passing `source=/dev/sdX` and `lvm.vg_name=bla` will let `LXD`:
- check whether there is a physical volume on `/dev/sdX`. if so, LXD will use it. if not, create a new physical volume on there.
- check whether a volume group `bla` exists on `/dev/sdX`. if so, LXD will use it. if not, create a new volume group there.
- after creation, `source` will be set to `bla` and thus be aligned with `lvm.vg_name` because the actual physical volume name `/dev/sdX` might change.

Recording lvm.vg_name sounds fine for the same reason that zfs.pool_name is fine.

stgraber commented:
@brauner sounds good. Is that what we do for ZFS too? I've noticed that none of my machines seem to have zfs.pool_name set after upgrade to 2.9.

brauner commented:
@stgraber, updated:
- implemented the patch to change from `volume.lvm.thinpool_name` to `lvm.thinpool_name`
- updated documentation to mention new keys
- introduced new key `lvm.vg_name`

brauner commented:
@stgraber, checking on the zfs case in a bit.

brauner commented:
jenkins: test this please.

brauner commented:
jenkins: test this please
